test/testdrive-old-kafka-src-syntax/mzcompose.py

```diff
@@ -237,3 +259,204 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
+    )
+    parser.add_argument(
+        "--kafka-default-partitions",
+        type=int,
+        metavar="N",
+        help="set the default number of kafka partitions per topic",
+    )
+    parser.add_argument(
+        "--system-param",
+        type=str,
+        action="append",
+        nargs="*",
+        help="System parameters to set in Materialize, i.e. what you would set with `ALTER SYSTEM SET`",
+    )
+
+    parser.add_argument("--replicas", type=int, default=1, help="use multiple replicas")
+
+    parser.add_argument(
+        "--default-timeout",
+        type=str,
+        help="set the default timeout for Testdrive",
+    )
+    parser.add_argument(
+        "files",
+        nargs="*",
+        default=["*.td"],
+        help="run against the specified files",
+    )
+
+    (args, _) = parser.parse_known_args()
+
+    matching_files = []
+    for filter in args.files:
+        matching_files.extend(
+            glob.glob(filter, root_dir="test/testdrive-old-kafka-src-syntax")
+        )
+    matching_files = [file for file in matching_files if file != "session.td"]
+
+    dependencies = [
+        "fivetran-destination",
+        "minio",
+        "materialized",
+        "postgres",
+        "mysql",
+    ]
+
+    if args.redpanda:
+        kafka_deps = ["redpanda"]
+    else:
+        kafka_deps = ["zookeeper", "kafka", "schema-registry"]
+
+    dependencies += kafka_deps
+
+    testdrive = Testdrive(
+        forward_buildkite_shard=True,
+        kafka_default_partitions=args.kafka_default_partitions,
+        aws_region=args.aws_region,
+        default_timeout=args.default_timeout,
+        volumes_extra=["mzdata:/mzdata"],
+        external_blob_store=True,
+        fivetran_destination=True,
+        fivetran_destination_files_path="/share/tmp",
+        entrypoint_extra=[f"--var=uses-redpanda={args.redpanda}"],
+    )
+
+    sysparams = args.system_param
+    if not args.system_param:
+        sysparams = []
+
+    additional_system_parameter_defaults = {}
+    for val in sysparams:
+        x = val[0].split("=", maxsplit=1)
+        assert len(x) == 2, f"--system-param '{val}' should be the format <key>=<val>"
+        key = x[0]
+        val = x[1]
+
+        additional_system_parameter_defaults[key] = val
+
+    mz_old = Materialized(
+        default_size=Materialized.Size.DEFAULT_SIZE,
+        image=get_old_image_for_source_table_migration_test(),
+        external_metadata_store=True,
+        external_blob_store=True,
+        additional_system_parameter_defaults=additional_system_parameter_defaults,
+    )
+
+    additional_system_parameter_defaults["force_source_table_syntax"] = "true"
+
+    mz_new = Materialized(
+        default_size=Materialized.Size.DEFAULT_SIZE,
+        image=get_new_image_for_source_table_migration_test(),
+        external_metadata_store=True,
+        external_blob_store=True,
+        additional_system_parameter_defaults=additional_system_parameter_defaults,
+    )
```

Comment:
@def- Does this look right? I forget in Python if this updates a shared reference or not.

Reply:
Nope, that's wrong, you need to copy the dict.
```suggestion
    mz_old = Materialized(
        default_size=Materialized.Size.DEFAULT_SIZE,
        image=get_old_image_for_source_table_migration_test(),
        external_metadata_store=True,
        external_blob_store=True,
        additional_system_parameter_defaults=dict(additional_system_parameter_defaults),
    )

    additional_system_parameter_defaults["force_source_table_syntax"] = "true"

    mz_new = Materialized(
        default_size=Materialized.Size.DEFAULT_SIZE,
        image=get_new_image_for_source_table_migration_test(),
        external_metadata_store=True,
        external_blob_store=True,
        additional_system_parameter_defaults=additional_system_parameter_defaults,
    )
```

Reply:
Thanks!